geopy/geocoders/herev7.py

```diff
@@ -0,0 +1,286 @@
+import collections.abc
+import warnings
+from functools import partial
+from urllib.parse import urlencode
+
+from geopy.exc import (
+    ConfigurationError,
+    GeocoderAuthenticationFailure,
+    GeocoderInsufficientPrivileges,
+    GeocoderQuotaExceeded,
+    GeocoderServiceError,
+    GeocoderUnavailable,
+)
+from geopy.geocoders.base import DEFAULT_SENTINEL, Geocoder
+from geopy.location import Location
+from geopy.util import join_filter, logger
+
+__all__ = ("HereV7", )
+
+
+class HereV7(Geocoder):
+    """Geocoder using the HERE Geocoder API.
+
+    Documentation at:
+        https://developer.here.com/documentation/geocoding-search-api/
+    """
+
+    structured_query_params = {
+        'street',
+        'houseNumber',
+        'postalCode',
+        'city',
+        'district',
+        'county',
+        'state',
+        'country'
+    }
+
+    geocode_path = '/v1/geocode'
+    reverse_path = '/v1/revgeocode'
+
+    def __init__(
+            self,
+            *,
+            apiKey=None,
```

Comment:
In Python arguments in public interfaces should adhere to snake_case. Could you please rename that to `apikey` or `api_key`?

Reply:
Fixed in 078c382